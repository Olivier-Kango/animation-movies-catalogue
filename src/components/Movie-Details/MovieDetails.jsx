import React, { useEffect } from 'react';
// import { Link } from 'react-router-dom';
import { HashLink } from 'react-router-hash-link';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMovies } from '../../redux/movies/movies';

const MovieDetails = () => {
  const url = window.location.href;
  const path = url.split('/')[3];
  const movies = useSelector((state) => state.movies);
  const dispatch = useDispatch();
  useEffect(() => { if (movies.length === 0) { dispatch(fetchMovies()); } }, []);

  const movie = movies.filter((movie) => movie.id === Number(path))[0];

  return (
    <div id={movie.id} key={`${movie.id}`}>
      <h2>{movie.original_title}</h2>
      <img src={`https://image.tmdb.org/t/p/w500${movie.backdrop_path}`} alt={`${movie.original_title}`} />
      <p>
        <strong>Popularity: </strong>
        {movie.popularity}
      </p>
      <p>
        <strong>Vote count: </strong>
        {movie.vote_count}
      </p>
      <p>
        <strong>Original language: </strong>
        {movie.original_language}
      </p>
      <p>
        <strong>Overview: </strong>
        {movie.overview}
      </p>
      <p>
        <strong>Realease date: </strong>
        {movie.release_date}
      </p>
      <br />
      <HashLink to={`/#${movie.id}`} smooth>
        <button type="button">Back</button>
      </HashLink>
    </div>
  );
};

export default MovieDetails;
