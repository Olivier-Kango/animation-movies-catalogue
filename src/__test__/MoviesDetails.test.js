import { render } from '@testing-library/react';
import {
  BrowserRouter as Router,
  Routes,
  Route,
} from 'react-router-dom';
import MovieDetails from '../components/Movie-Details/MovieDetails';

describe('Testing the MoviesDetails page:', () => {
  test('Should render', () => {
    render(
      <Router>
        <Routes>
          <Route path="/:id" element={<MovieDetails />} />
        </Routes>
      </Router>
    );
  });
  it('fectch Movies Details', () => {
    const { container } = render(
      <Router>
        <Routes>
          <Route path="/:id" element={<MovieDetails />} />
        </Routes>
      </Router>
    );
    container.querySelector('.details');
    expect(container.firstChild).toMatchSnapshot();
  });
});
