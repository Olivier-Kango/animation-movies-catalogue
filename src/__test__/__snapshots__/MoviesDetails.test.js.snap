// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the MoviesDetails page: fectch Movies Details 1`] = `null`;
